tests/stdlib/tchannels.nim

```diff
@@ -0,0 +1,26 @@
+discard """
+  matrix: "--gc:orc --threads:on -d:danger"
+"""
+
+import std/[channels, isolation]
+
+var
+  sender: array[10, Thread[void]]
+  receiver: array[5, Thread[void]] 
+
+var chan = newChannel[seq[string]](40)
+proc sendHandler() =
+  chan.send(isolate(@["Hello, Nim"]))
+proc recvHandler() =
+  var x = chan.recv()
+  discard x
+
+template benchmark() =
+  for t in mitems(sender):
+    t.createThread(sendHandler)
+  joinThreads(sender)
+  for i in 0 .. receiver.high:
+    createThread(receiver[i], recvHandler)
+  joinThreads(receiver)
+
+benchmark()
```

Comment:
this test case is not good because it already worked before https://github.com/nim-lang/Nim/pull/17391 (with this modification because of recent API change:
```nim
    var x = chan.recv()
    =>
    var x: seq[string]
    chan.recv(x)
```

instead use this:
```nim
discard """
timeout:  5.0 # but typically < 1s
"""

when true:
  # bug #17380: this was either blocking or crashing with SIGSEGV
  import std/[channels, isolation]
  const
    N1 = 10
    N2 = 100
  var
    sender: array[N1, Thread[void]]
    receiver: array[5, Thread[void]] 

  var chan = newChannel[seq[string]](N1 * N2) # large enough to not block
  proc sendHandler() =
    chan.send(isolate(@["Hello, Nim"]))
  proc recvHandler() =
    var x: seq[string]
    chan.recv(x)
    discard x

  template benchmark() =
    for t in mitems(sender):
      t.createThread(sendHandler)
    joinThreads(sender)
    for t in mitems(receiver):
      t.createThread(recvHandler)
    joinThreads(receiver)
  for i in 0..<N2:
    benchmark()
```

note that this will timeout prior to https://github.com/nim-lang/Nim/pull/17391; if needed we can also add a test that would SIGSEGV prior to https://github.com/nim-lang/Nim/pull/17391 but work (and not timeout) after https://github.com/nim-lang/Nim/pull/17391, but it's a bit tricker

Reply:
done, thanks